### PR TITLE
Dropdown component

### PR DESCRIPTION
**Pull request by bpasha7**

### CODE REVIEW

File: src/states/DropdownState.vue [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        },
+        disable: {
+            type: Boolean,
+            default: () => false
+        },
+        options: {
+            type: Array,
+            default: () => []
+        },
+        multiply: {
```
Comment (trueromanus) [inline src/states/DropdownState.vue:22]: May be multiple?

File: src/states/DropdownState.vue [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        },
+        validateHost: {
+            type: Object
+        }
+    },
+    data () {
+        return {
+            open: false,
+            isValid: false,
+            selectedOptions: null
```
Comment (trueromanus) [inline src/states/DropdownState.vue:42]: May be new Set() by default. You don't need create it after it.

File: src/states/DropdownState.vue [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                    if (this.validateHost) this.validateHost.add(this, { rule: key, messages: value.messages });
+                }
+            }
+        },
+        toggle() {
+            this.open = !this.open;
+        },
+        setSelectedOptions(newValue) {
+            this.selectedOptions = null;
+            const emptyMap = new Set();
```
Comment (trueromanus) [inline src/states/DropdownState.vue:67]:
```js
this.selectedOptions = newValue ? new Set(newValue) : new Set();
```

File: src/examples/helloworld/App.vue [hunk 4]
```diff
@@ -146,6 +177,7 @@
 //load bootstrap styles
 require.resolveStyles(`https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta2/dist/css/bootstrap.min.css`);
 require.resolveStyles(`https://unpkg.com/material-components-web@latest/dist/material-components-web.min.css`);
+// await require.resolveScripts(`https://unpkg.com/material-components-web@latest/dist/material-components-web.min.css`);
```
Comment (trueromanus) [inline src/examples/helloworld/App.vue:180]: Dead code

File: src/states/DropdownState.vue (hunk 1 above)
Comment (bpasha7) [reply src/states/DropdownState.vue]: renamed

File: src/states/DropdownState.vue (hunk 2 above)
Comment (bpasha7) [reply src/states/DropdownState.vue]: done

File: src/examples/helloworld/App.vue (hunk 4 above)
Comment (bpasha7) [reply src/examples/helloworld/App.vue]: removed

File: src/states/DropdownState.vue (hunk 3 above)
Comment (bpasha7) [reply src/states/DropdownState.vue]: refactored

**Pull request merged**